app/src/main/java/com/example/seminar1/FollowerAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.seminar1
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.seminar1.databinding.ItemListBinding
+
+class FollowerAdapter :  RecyclerView.Adapter<FollowerAdapter.FollowerViewHolder>(),ItemTouchHelperCallback.ItemTouchHelperListener{
+    val userList = mutableListOf<UserData>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FollowerViewHolder {
+        val binding =
+                ItemListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return FollowerViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: FollowerViewHolder, position: Int) {
+        holder.onBind(userList[position])
+
+        holder.itemView.setOnClickListener {
+            itemClickListener.onClick(it,position)
+        }
+    }
+
+    override fun getItemCount(): Int = userList.size
+
+    class FollowerViewHolder(
+            private val binding: ItemListBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun onBind(data: UserData) {
+            binding.tvName.text = data.name
+            binding.tvIntroduce.text = data.introduction
+
+
+        }
+    }
+
+    //드래그 호출 메소드
+    override fun onItemMove(from_position: Int, to_position: Int) : Boolean {
+        val user = userList[from_position]
+
+        userList.removeAt(from_position)
+        userList.add(to_position,user)
+
+        notifyItemMoved(from_position, to_position)
+        return true
+    }
+
+    //스와이프 호출 메소드
+    override fun onItemSwipe(position: Int){
+        userList.removeAt(position)
+        notifyItemRemoved(position)
+    }
+
+
+    //클릭 인터페이스 정의
+    interface ItemClickListener{
```

Comment:
```suggestion
    fun interface ItemClickListener {
```
코틀린에 추가된 문법으로 `interface` 앞에 `fun` 을 붙여 SAM(Single Abstract Method) 를 지원하는 인터페이스로 만들 수 있습니다. 이렇게 되면 인터페이스의 익명 객체를 만들 때 바로 람다로 사용이 되는 것으로 압니다.

```kotlin
    fun interface A {
        fun hi()
    }
    fun B(a: A) {}
    B { } // No problem
```